packages/core/src/_common/base/api_middleware.js

```diff
@@ -20,19 +105,37 @@ class APIMiddleware {
         const promise = promiseRejectToResolve(response_promise);
 
         const key = requestToKey(request);
+        const request_type = this.getRequestType(request);
+
+        if (request_type) performance.mark(`${request_type}_start`);
 
         if (options.callback) {
             promise.then(options.callback);
         }
 
         this.debounced_calls[key] = promise;
 
-        promise.then(() => {
+        promise.then(response => {
+            const response_type = this.getRequestType(response);
+            if (response_type) this.defineMeasure(response_type);
```

Comment:
looks fine, but what do you think on the possibility of having a proxy for `sendIsCalled` inside p2p package. So that the logic of datadog and functions like defineMeasure and others will run only when user visits p2p package. If possible then we can save the extra logic been run for other packages.

Reply:
i have added performance marks which are only called for specific scenarios in p2p and only the measures of those will be sent to datadog